### PR TITLE
Fix the issue of unexpected behavior related to comments inside entities.

### PR DESCRIPTION
**Pull request by sahanHe**
## Purpose

Fixes: [3591](https://github.com/ballerina-platform/ballerina-standard-library/issues/3591)

## Checklist
- [x] Linked to an issue
- [ ] ~Updated the specification~
- [ ] ~Updated the changelog~
- [x] Added tests

### CODE REVIEW

File: persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_29/Ballerina.toml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[package]
+org = "foo"
+name = "persist_generate_29"
+version = "0.1.0"
+distribution = "2201.3.0"
+
+[build-options]
+observabilityIncluded = false
+
+
```
Comment (daneshk) [inline persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_29/Ballerina.toml:10]:
```suggestion
```

File: persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_27/entities.bal [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    @persist:Relation {keyColumns: ["profileId"], reference: ["id"]}
+    //comment11
+    Profile profile?;
+    //comment8
+    @persist:Relation {keyColumns: ["userId"], reference: ["id"]}
+    User user?;
+|};
+
+
+
```
Comment (daneshk) [inline persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_27/entities.bal:63]:
```suggestion
```

File: persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_29/Ballerina.toml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+[package]
+org = "foo"
+name = "persist_generate_29"
+version = "0.1.0"
+distribution = "2201.3.0"
+
+[build-options]
+observabilityIncluded = false
+
+
```
Comment (daneshk) [inline persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_29/Ballerina.toml:10]:
```suggestion
```

File: persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java [hunk 4]
```diff
@@ -281,6 +283,27 @@ public static void populateEntities(Module.Builder moduleBuilder, SyntaxTree bal
...
+            case QUALIFIED_NAME_REFERENCE:
+                QualifiedNameReferenceNode qualifiedName = (QualifiedNameReferenceNode) typeDesc;
+                String modulePrefix = qualifiedName.modulePrefix().text();
+                String identifier = qualifiedName.identifier().text();
+                String qualifiedType = modulePrefix + COLON + identifier;
+                return qualifiedType;
+            case SIMPLE_NAME_REFERENCE:
+                return ((SimpleNameReferenceNode) typeDesc).name().text();
+            default:
+                return "";
```
[the remark this answers is not in the document]
Comment (sahanHe) [reply persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java]: Updated in [8458b4a](https://github.com/ballerina-platform/persist-tools/pull/84/commits/8458b4a70f4b0fd2252b20bd726f50600b0dda06)

File: persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_29/Ballerina.toml (hunk 3 above)
Comment (sahanHe) [reply persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_29/Ballerina.toml]: Updated in [8458b4a](https://github.com/ballerina-platform/persist-tools/pull/84/commits/8458b4a70f4b0fd2252b20bd726f50600b0dda06)

File: persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_27/entities.bal (hunk 2 above)
Comment (sahanHe) [reply persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_27/entities.bal]: Updated in [8458b4a](https://github.com/ballerina-platform/persist-tools/pull/84/commits/8458b4a70f4b0fd2252b20bd726f50600b0dda06)

File: persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_29/Ballerina.toml (hunk 1 above)
Comment (sahanHe) [reply persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_29/Ballerina.toml]: Updated in [8458b4a](https://github.com/ballerina-platform/persist-tools/pull/84/commits/8458b4a70f4b0fd2252b20bd726f50600b0dda06)

**Review by daneshk — APPROVED** [review]

**Pull request merged**